rfcs/convergence/react-focus-management.md

```diff
@@ -0,0 +1,169 @@
+
+```typescript
+// Basic up/down keyboard navigation
+const ref = useKeyboardNavigationGroup();
+
+<div ref={ref}>{props.children}</div>;
+```
+
+```typescript
+const ref = React.useReact<HTMLElement>();
+const {
+  findFirst,
+  findLast,
+  findAll,
+} = useFocusable(ref); // All operations are with respect to the given DOM container
+
+findFirst() // first focusable
+findLast() // first last focusable
+findAll(condition: () => true) // first all focusables with condition
+
+<div ref={ref}>{props.children}</div>
+```
+
+#### useFocusable
+
+A hook that exposes helper methods to traverse focusable elements in the context of a DOM element.
+
+The below table lists the helper functions to support:
+
+| Name      | Value         | Options                  | Description                                         |
+| --------- | ------------- | ------------------------ | --------------------------------------------------- |
+| findFirst | HTMLElement   |                          | Finds the first focusable element                   |
+| findLast  | HTMLElement   |                          | Finds the last focusable element                    |
+| findAll   | HTMLElement[] | boolean matcher function | Finds all focusable elements that match a condition |
+
+#### useKeyboardNagiationGroup
+
+A hook that returns a `ref` to be assigned to the container of a group of focusable elements that should support keyboard navigation
+
+The below table lists the configuration options that the hook should support:
+
+| Name          | Value                  | Description                                                                                               |
+| ------------- | ---------------------- | --------------------------------------------------------------------------------------------------------- |
+| axis          | vertical \| horizontal | Indicates which direction keyboard navigation should use                                                  |
+| enableHomeEnd | boolean                | Enables the use of `Home` and `End` keys to navigate to first and last elements of the group respectively |
+| cyclic        | boolean                | Indicates that navigation should not stop at first and last elements respectively                         |
+
+### Pros and Cons
+
+<!-- Enumerate the pros and cons of the proposal. Make sure to think about and be clear on the cons or drawbacks of this propsoal. If there are multiple proposals include this for each. -->
+
+#### Pros
+
+The current development pace of converged components means we will be able iterate quickly on this API since components under development such as `Menu`, `Accordion` and `Tooltip` will require these behaviours in the coming project cycles
```

Comment:
```suggestion
The current development pace of converged components means we will be able iterate quickly on this API since components under development such as `Menu`, `Accordion` and `Tooltip` will require these behaviours in the coming project cycles.
```

If `Tooltip` is mentioned, do we have a proposal for focus traps?

Reply:
I'm currently working on one, I can try add it to this RFC we'll see how it goes